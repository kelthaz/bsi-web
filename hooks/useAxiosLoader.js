import { useCallback, useEffect, useMemo, useState } from 'react';
import axiosIntance from '../config/AxiosConfig';

const useAxiosLoader = () => {
  const [counter, setCounter] = useState(0);
  const inc = useCallback(() => setCounter((counterInt) => counterInt + 1), [setCounter]);
  const dec = useCallback(() => setCounter((counterInt) => counterInt - 1), [setCounter]);

  const interceptors = useMemo(
    () => ({
      request: (config) => {
        inc();
        return config;
      },
      response: (response) => {
        dec();
        return response;
      },
      error: (error) => {
        dec();
        return Promise.reject(error);
      },
    }),
    [inc, dec]
  );

  useEffect(() => {
    const reqInterceptor = axiosIntance.interceptors.request.use(interceptors.request, interceptors.error);
    const resInterceptor = axiosIntance.interceptors.response.use(interceptors.response, interceptors.error);
    return () => {
      axiosIntance.interceptors.request.eject(reqInterceptor);
      axiosIntance.interceptors.response.eject(resInterceptor);
    };
  }, [interceptors]);

  return [counter > 0, inc, dec];
};

export default useAxiosLoader;
