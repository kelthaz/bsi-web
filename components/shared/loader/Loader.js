import styles from './loader.module.scss';
import SvgLogoBanCoppelPymes from '../../svgs/logos/SvgLogoBanCoppelPymes';
import useAxiosLoader from '../../../hooks/useAxiosLoader';

const Loader = () => {
  const [loading] = useAxiosLoader();

  return (
    loading && (
      <div className={styles['container-loader']}>
        <div className={styles.spinner} />
        <SvgLogoBanCoppelPymes />
      </div>
    )
  );
};

export default Loader;
