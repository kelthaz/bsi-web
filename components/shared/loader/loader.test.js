import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import useAxiosLoader from '../../../hooks/useAxiosLoader';
import Loader from './Loader';

jest.mock('../../../hooks/useAxiosLoader', () => jest.fn());
describe('Pruebas en el componente Loader', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Debe de mostrarse correctamente el componente', () => {
    // arrange
    useAxiosLoader.mockReturnValue([true]);
    // act
    const wrapper = shallow(<Loader />);
    // assert
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de no mostrarse el componente', () => {
    // arrange
    useAxiosLoader.mockReturnValue([false]);
    // act
    const wrapper = shallow(<Loader />);
    // assert
    expect(wrapper.find('div').exists()).toBe(false);
  });
});
