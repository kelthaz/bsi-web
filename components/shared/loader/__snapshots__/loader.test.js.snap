// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Loader Debe de mostrarse correctamente el componente 1`] = `
<div
  className="container-loader"
>
  <div
    className="spinner"
  />
  <SvgLogoBanCoppelPymes />
</div>
`;
